aisim/atoms.py

```diff
@@ -87,6 +87,25 @@ def state_vectors(self, new_state_vectors):
             assert new_state_vectors.shape[0] == len(self)
             self._state_vectors = new_state_vectors
 
+    @property
+    def density_matrices(self):
+        """
+        n × m x m array, representing the density matrix of the m level system of the n atoms.
+        These are pure states.
+        """
+        # |Psi><Psi|
+        return np.einsum('ji,il->ijl', np.conjugate(self.state_vectors).T, self.state_vectors)

```

Comment:
neat, seems to work as intended

Reply:
yes, einsum might become my new favourite numpy function. it can be used for matrix multiplication, trace, sum over columns/rows and much more. really cool